shared/fs/header/index.js

```diff
@@ -0,0 +1,7 @@
+// @flow
+import FolderHeader from './container'
```

Comment:
Why is this file necessary? I.e. why can't callers just import from `header/container`?

Reply:
Good point. I'm gonna reorganize these a bit after these PRs are merged. Fred also suggested some naming schemes above.

Reply:
We've been doing a lot of renaming lately. It creates conflicts when we're working on the same files. Let's try to minimize how often this has to happen.

Reply:
Some of the renamings here are because some types are getting bloated and I wanted to split them up.